src/subscriptions/firebase.js

```diff
@@ -1,66 +1,26 @@
-import * as Sentry from '@sentry/browser'
 import firebaseConfig from './firebase.config'
 
-const SERVICE_WORKER_PATH = '/service-worker.js'
-const SERVICE_WORKER_SCOPE = '/karrot-push'
-
-export async function hasServiceWorker () {
-  return (await getServiceWorkers()).length > 0
-}
-
-export async function getServiceWorker () {
-  if (await hasServiceWorker()) return (await getServiceWorkers())[0]
+export function isSupported () {
+  return Boolean(navigator.serviceWorker)
 }
 
-async function getServiceWorkers () {
-  if (__ENV.CORDOVA || !window.navigator.serviceWorker) return []
-  let registrations
-  try {
-    registrations = await window.navigator.serviceWorker.getRegistrations()
-  }
-  catch (err) {
-    Sentry.captureException(err)
-    return []
-  }
-  return registrations.filter(worker => worker.scope.endsWith(SERVICE_WORKER_SCOPE))
-}
-
-async function getOrCreateWorker () {
-  const worker = await getServiceWorker()
-  if (worker) return worker
-  return window.navigator.serviceWorker.register(SERVICE_WORKER_PATH, { scope: SERVICE_WORKER_SCOPE })
+export async function getServiceWorkerRegistration () {
+  // See also https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerContainer/ready
+  return await navigator.serviceWorker.ready
 }
 
 let messaging
 export async function initializeMessaging () {
-  const { initializeApp, messaging: initializeMessaging } = await import('./firebase.lib')
   if (messaging) return messaging
 
-  initializeApp(firebaseConfig)
-  messaging = await initializeMessaging()
-  messaging.useServiceWorker(await getOrCreateWorker())
-  return messaging
-}
+  const { initializeApp, messaging: messagingFactory } = await import('./firebase.lib')
 
-/*
- * When we disable push notifications we need to remove the service workers.
- * If we remove them immediately, then re-enable push notifications then
- * firebase messsaging library gets upset. So we just remember whether we want them removed
- * only do it when the page actually closes.
- */
+  const app = initializeApp(firebaseConfig)
+  messaging = await messagingFactory(app)
 
-let shouldRemoveServiceWorkersOnUnload = false
-export async function removeServiceWorkersOnUnload (value) {
-  shouldRemoveServiceWorkersOnUnload = value
-}
+  messaging.onMessage(m => {
+    console.log('onMessage', m)
+  })
 
-async function removeServiceWorkers () {
-  for (const worker of await getServiceWorkers()) {
-    worker.unregister()
-  }
-}
-
-window.onbeforeunload = () => {
-  // Do not return anything from this function or it triggers a browser dialog
-  if (shouldRemoveServiceWorkersOnUnload) removeServiceWorkers()
```

Comment:
My goal with all this stuff was to not leave service workers hanging around if we don't actually need one. They seem like pollution to me. If they don't have push enabled, or any offline processing why have one registered?

That said, I'm fine with the new approach to always have one.

Reply:
It's not that the user enables offline processing, it'll be a default now - in a way, having service workers seems the fancy version of browser caching. We set our assets to "max expiry" and don't care how long they will be on users machines, but the browser takes care of that. I would expect browsers to occasionally clear old service workers if they are not needed anymore - not sure if they do that already...

Reply:
I mean, they are _never_ needed at the moment on the desktop site if you don't enable push notifications, but will be there anyway. As I said though, I'm fine to go with having them always enabled, I don't have any compelling evidence that it's beneficial to minimise their use.